core/beneficiary/saver.go

```diff
@@ -61,23 +63,77 @@ type l1Saver struct {
 	set     settler
 	ad      addressProvider
 	chainID int64
+	beneficiaryChangeKeeper
 }
 
-func newL1Saver(chainID int64, ad addressProvider, bc multiChainBC, set settler) *l1Saver {
+func newL1Saver(chainID int64, ad addressProvider, set settler, st storage) *l1Saver {
 	return &l1Saver{
-		chainID: chainID,
-		set:     set,
-		ad:      ad,
+		chainID:                 chainID,
+		set:                     set,
+		ad:                      ad,
+		beneficiaryChangeKeeper: beneficiaryChangeKeeper{st: st, chainID: chainID},
 	}
 }
 
+// SettleWithBeneficiaryState represents the state of settle with beneficiary transaction
+type SettleWithBeneficiaryState string
+
+const bucketBeneficiaryChangeStatus = "beneficiary-change-status"
+const (
+	// PENDING transaction is pending
+	PENDING SettleWithBeneficiaryState = "pending"
+	// COMPLETED transaction is completed
+	COMPLETED SettleWithBeneficiaryState = "completed"
+)
+
+type beneficiaryChangeKeeper struct {
+	st      storage
+	chainID int64
+}
+
+// BeneficiaryChangeStatus holds Beneficiary settlement transaction information
+type BeneficiaryChangeStatus struct {
+	State SettleWithBeneficiaryState
+	Error string
+}
+
+func (bcs *beneficiaryChangeKeeper) updateBeneficiaryChangeStatus(id identity.Identity, sbs SettleWithBeneficiaryState, settleError error) {
+	var errorMSG string
+	if settleError != nil {
+		errorMSG = settleError.Error()
+	}
+	err := bcs.st.SetValue(bucketBeneficiaryChangeStatus, storageKey(bcs.chainID, id.Address), &BeneficiaryChangeStatus{
+		State: sbs,
+		Error: errorMSG,
+	})
+
+	log.Err(err).Msg("Failed to update BeneficiaryChangeStatus")
+}
+
+// BeneficiaryChangeStatus get beneficiary change status for given identity
+func (bcs *beneficiaryChangeKeeper) BeneficiaryChangeStatus(id identity.Identity) *BeneficiaryChangeStatus {
```

Comment:
There are few things I would consider:
* Add `Get` or something like that in front of the function name to indicate what exactly it's doing
* Dont return possible `nil` values. Either return `(*BeneficiaryChangeStatus, bool)` or `(*BeneficiaryChangeStatus, error)` and possibly drop loging from here. From general practice, returning `nil` and no error results in unwanted panics :(